### DISCUSSION TITLE
Procedure entry point__kmpc_free could not be located in the dynamic link library

### DISCUSSION THREAD

**Discussion by FGarelli** [OpenRadioss/OpenRadioss#1452] [Q&A]
Hi, i built the starter and the engine of openradioss but i cannot launch any job cause of this error: Procedure entry point__kmpc_free could not be located in the dynamic link library C:\tmp\openradioss\exec\starter_win64.exe
Any ideas of how to solve it?

**Comment by OlivierWienholtz** [comment]
Hi FGarelli,

Radioss has a dynamic link. Including the Intel Fortran runtime libraries.
Please 

execute the Starter in a shell with the Intel compilers Loaded.

or you can also call in your shell  - assuming you have a standard Compiler installation.
call "c:\Program Files (x86)\Intel\oneAPI\setvars.bat"


Take care : other environment variables needs to be set to execute properly OpenRadioss. 
See : https://github.com/OpenRadioss/OpenRadioss/blob/main/INSTALL.md

Thanks

**Reply by FGarelli** [reply]
Hi Olivier,
I'm using the following code to run the starter but i continue to receive the same error (Procedure entry point__kmpc_free could not be located in the dynamic link library). In the directory C:/tmp/test are present the files that i wanna run as a test:

cd C:\tmp\test
set OPENRADIOSS_PATH=C:\tmp\OpenRadioss
set RAD_CFG_PATH=%OPENRADIOSS_PATH%\hm_cfg_files
set RAD_H3D_PATH=%OPENRADIOSS_PATH%\extlib\h3d\lib\win64
set KMP_STACKSIZE=400m
set PATH=%OPENRADIOSS_PATH%\extlib\hm_reader\win64;%PATH% 
call "C:\Program Files (x86)\Intel\oneAPI\setvars.bat" intel64 vs2022
set OMP_NUM_THREADS=1
C:\tmp\OpenRadioss\exec\starter_win64.exe -i BIRD_WINDSHIELD_v1_0000 -np 1
C:\tmp\OpenRadioss\exec\engine_win64.exe  -i BIRD_WINDSHIELD_v1_0001

Any suggestion? I tried both with mpi and without but i received the same errors.

**Reply by FGarelli** [reply]
<img width="541" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/132582318/cca29d08-adb6-47b1-9df0-f5e7c6651bdd">

**Reply by MarianBulla** [reply]
Hi FGarelli,
can you try:
C:\tmp\OpenRadioss\exec\starter_win64.exe -i BIRD_WINDSHIELD_v1_0000.rad
C:\tmp\OpenRadioss\exec\engine_win64.exe -i BIRD_WINDSHIELD_v1_0001.rad
please ?

**Reply by FGarelli** [reply]
Hi Marian,
I tried but i receive the same error unfortunally

**Reply by MarianBulla** [reply]
Can you doubleckeck, if you have the same input, as me, please: 
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/ec437868-7f84-48d6-88cc-c70a654baa92)

**Comment by FGarelli** [comment]
Here is the only difference, on the intelpython
<img width="465" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/132582318/1210172c-cb99-46e6-a71a-9f8e415503c4">
Here the complete code that i runned as the one that you post.
<img width="464" alt="image" src="https://github.com/OpenRadioss/OpenRadioss/assets/132582318/c3b35bd8-36f3-42c3-8e09-81d787d9073e">

**Comment by OlivierWienholtz** [comment]
Hi 

__kmpc_free is part of the OpenMP layer in OneMPI : it  is in the libiomp5md.dll
libopmp5md.dll is a library shipped with the Intel Compiler. Radioss Starter & Engine use it.

In my installation : this library is in : C:\Program Files (x86)\Intel\oneAPI\compiler\2023.1.0\windows\redist\intel64_win\compiler
(I use currently OneAPI 2023.1)

During compiler Installation this directory should have been created, & this file libiomp5md.dll should be in.
Furthermore : during your compiler loading : the variable PATH should have this directory in.

I could not figure out in which Package from which toolkit this library is in.

Maybe you can check if you find this file as 64 bit in your OneAPI Installation,
and then check if %PATH% is correct.

---

Another Possibility, is that the Starter found another library named libiomp5md.dll from an older Intel runtime like the one shipped with OpenRadioss releases (it is a slight different older OneAPI version)

Can you check if you have this library in your test example directory for instance.
Can you check if you have some instances of this lib somewhere else in C:
